components/brave_wallet/renderer/js_ethereum_provider.cc

```diff
@@ -155,6 +156,14 @@ void JSEthereumProvider::CreateEthereumObject(v8::Isolate* isolate,
   v8::Local<v8::Object> global = context->Global();
   v8::Local<v8::Object> ethereum_obj;
   v8::Local<v8::Object> metamask_obj;
+
+  blink::ScriptState* script_state = blink::ScriptState::From(isolate->GetCurrentContext());
```

Comment:
this is illegal to access here

Reply:
I guess you can only go from ScriptState to ExecutionContext and not the other way around, but you can pass `isolate->GetCurrentContext()`